pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -1213,6 +1213,15 @@ func IsTCPEnforcedForNodeLocalDNSToUpstreamDNS(systemComponents *gardencorev1bet
 	return fromSpec && fromAnnotation
 }
 
+// IsDisabledNodeLocalDNSForwardToUpstreamDNS DisableForwardToUpstreamDNS indicates whether requests from node local DNS to upstream DNS is disabled.
+// It can be disabled via the annotation (legacy) or via the shoot specification.
+func IsDisabledNodeLocalDNSForwardToUpstreamDNS(systemComponents *gardencorev1beta1.SystemComponents) bool {
+	if systemComponents != nil && systemComponents.NodeLocalDNS != nil && systemComponents.NodeLocalDNS.DisableForwardToUpstreamDNS != nil {
+		return *systemComponents.NodeLocalDNS.DeepCopy().DisableForwardToUpstreamDNS
```

Comment:
Why do we need the `DeepCopy()` here? We don't do this in other helper functions, do we?